Neos.Fusion/Classes/FusionObjects/CollectionImplementation.php

```diff
@@ -18,6 +18,7 @@
  *
  * //tsPath collection *Collection
  * //tsPath itemRenderer the TS object which is triggered for each element in the node collection
+ * @deprecated since Neos 4.2 in favor of IterationImplementation
```

Comment:
Would be good to name the version, where it will be removed (5.0 ?)

Reply:
I am not sure. I think we should allow some time util we remove those prototypes since.

Reply:
I would wait until version 6 since it is a massive change. Especially for package developer: They would have to handle multiple versions. If we remove it later, the transition would be smoother

Reply:
Doesn't a simple renaming code migration solve the major issue? AFAIS the usages are fully b/c otherwise apart from the namechange, no?